### DISCUSSION TITLE
My FastAPI Python project is classified as Jupyter Notebook

### DISCUSSION THREAD

**Discussion by mrisher** [github-linguist/linguist#6618] [Classification]
In my private repository (https://github.com/goplayfully/gameserver) the bulk of the activity (400 commits) are on the FastAPI Python files, but for some reason the repository is detected as Jupyter Notebook. Even more surprising, when I click on "Jupyter" -- which is listed as 79% -- I get only one single file, whereas "Python" (19%) has 70 files.  Jupyter defaults to the extension `.ipynb` whereas Python has to be the more common use of `.py` right? 

I wonder if this has to do with 19 pull requests being identified as Jupyter? 

I haven't tried overriding with `.gitattributes` because your troubleshooter led me here first. It's not a major issue but I'm super curious why it's wrong and how to correct it.

Thanks!

![image](https://github.com/github-linguist/linguist/assets/218038/0559505c-d042-49ab-8a03-226f32e819a3)
 
![image](https://github.com/github-linguist/linguist/assets/218038/22a8752c-a55a-455e-b712-39411dbc910f)

![image](https://github.com/github-linguist/linguist/assets/218038/2ff151df-662d-4807-8b35-ece772797afa)

**Comment by lildude** [comment]
This is the reason you're seeing this behaviour is explained in the [How Linguist works](https://github.com/github-linguist/linguist/blob/master/docs/how-linguist-works.md) doc:

> The percentages are calculated based on the bytes of code for each language as reported by the [List Languages](https://docs.github.com/rest/reference/repos#list-repository-languages) API.

Your Jupyter file is going to be much larger in bytes of code than all your Python files combined.

**Reply by mrisher** [reply]
That makes sense but it's not larger than the Python files *combined* -- the Jupyter file is 16kb, which is bigger than individual Python files but not than the accumulation (there are a handful of 15kb+ Python files). My intuition would be to use those cumulative byte counts -- or a tally of files, grouped by type -- rather than a single outlier file.

But you've at least given me a good idea how to override it, so thank you for that 👍🏽

**Reply by lildude** [reply]
If you grant me access to your repo, I can provide you a more precise explanation as to what is happening... it's possible your files are falling into locaton Linguist ignores because the path is considered generated, vendored or documentation.

**Reply by lildude** [reply]
NM. From GitHub's back-end, I can see your Jupyter files total 1.04 MB whilst your Python files total 263 KB (I can't see your actual files or content). It's likely one or more of your files is larger than 350kb so GitHub's search isn't indexing them, hence you're not seeing them when you click the language. See https://docs.github.com/en/search-github/github-code-search/about-github-code-search#limitations for the Search limitations.

**Comment by mrisher** [comment]
Aha, that solves it, thank you so much! 
There are a few larger notebooks that didn't show up in the index, and those are cumulatively larger. You're right! 
Thank you for the help!

```
36438480     1056 -rw-r--r--    1 mrisher          staff              538231 Oct 25 14:18 .//ml_playground/topic_modelling/topic_modelling.ipynb
19093469      728 -rw-r--r--    1 mrisher          staff              368824 Jul 29 14:44 .//submodules/trivia/venv/lib/python3.10/site-packages/chromadb/experimental/density_relevance.ipynb
21792882       40 -rw-r--r--    1 mrisher          staff               16710 Sep  1 10:22 .//colab/Parse_Travel_Blog.ipynb
36438499     1056 -rw-r--r--    1 mrisher          staff              538231 Oct 25 14:18 .//src/topic_modelling/topic_modelling.ipynb
```
